packages/server/src/db/db.service.ts

```diff
@@ -0,0 +1,121 @@
+import { getRepository, Repository, IsNull } from 'typeorm';
+
+import UserEntity from '../models/user.entity';
+import EmailVerificationEntity from '../models/email-verification.entity';
+import PasswordResetEntity from '../models/password-reset.entity';
+import SocialProfileEntity from '../models/social-profile.entity';
+import SocialProfileDto from '../models/social-profile.dto';
+import RegistrationDto from '../models/registration.dto';
+import { ProvidersIdDBFieldName } from 'providers-auth/providers-auth.interfaces';
+
+interface CreateUserAuthEntityPayload {
+  userId: string;
+  expiresIn: number;
+}
+
+interface CreateEmailVerificationPayload extends CreateUserAuthEntityPayload {
+  emailVerificationToken?: string;
+  emailResetToken?: string;
+}
+
+interface CreatePasswordResetPayload extends CreateUserAuthEntityPayload {
+  passwordResetToken: string;
+}
+
+export default class DBService {
+  private userRepository: Repository<UserEntity>;
+  private emailVerificationRepository: Repository<EmailVerificationEntity>;
+  private passwordResetRepository: Repository<PasswordResetEntity>;
+  private socialProfileRepository: Repository<SocialProfileEntity>;
+
+  constructor() {
+    this.userRepository = getRepository(UserEntity);
+    this.emailVerificationRepository = getRepository(EmailVerificationEntity);
+    this.passwordResetRepository = getRepository(PasswordResetEntity);
+    this.socialProfileRepository = getRepository(SocialProfileEntity);
+  }
+
+  public async getUserById(id: string) {
+    return this.userRepository.findOne({ id });
```

Comment:
Shouldn't we use await before DB call?

Reply:
It is one call per method so I it is going to be `awaited` anyways

Reply:
Could you elaborate on this please?

Reply:
Well, `this.userRepository.findOne({ id })` returns promise. We do not need this promise result inside the method, it is simply returned, right? So the `user` is not needed to be `awaited` for inside the method. And outside the method we call `await getUserById(...)` meaning the program will anyways wait for the returned promise to be resolved.

Reply:
So getUserById method doesn't need to be async I guess.